tests/integration/block_mount_unmount_test.go

```diff
@@ -0,0 +1,296 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"testing"
+
+	"fmt"
+	"github.com/rook/rook/tests/framework/clients"
+	"github.com/rook/rook/tests/framework/contracts"
+	"github.com/rook/rook/tests/framework/installer"
+	"github.com/rook/rook/tests/framework/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+// ************************************************
+// ***Scenarios tested by the BlockMountUnMountSuite ***
+// Setup
+// - via the cluster CRD
+// - set up Block PVC - With ReadWriteOnce
+// - set up Block PVC - with ReadWriteMany
+// - Mount Volume on a pod and write some data
+// - UnMount Volume
+// Monitors
+// - one mons in the cluster
+// OSDs
+// - Bluestore running on directory
+// Block Mount & UnMount scenarios - repeat for each PVC
+// 1. ReadWriteOnce
+// 	  a. Mount Volume on a new pod - make sure persisted data is present and write new data
+//    b. Mount volume on two pods with  - mount should be successful only on first pod
+//2. ReadOnlyMany
+//   a. Mount Multiple pods with same volume - All pods should be able to read data
+//   b. ,Mount Multiple pods with same volume - All pods should not be able to write data
+//4. Run StatefulSet with PVC
+//	a. Scale up pods
+//  b. Scale down Pods
+//  c. Failover pods
+//  d . Delete statefulSet
+// ************************************************
+
+func TestBlockMountUnMountSuite(t *testing.T) {
```

Comment:
why do you need a new sweet for this? Seems like it should be part of the block suite

Reply:
For mount and unmont tests we do additional set up(eg creating various PVCs) which we dont while just creating block tests. Also as part of #1218 i will combine all this into smaller suites so there is less overhead of installign and uninstalling rook.